### ISSUE TITLE
About .txt files in the label folder

### ISSUE THREAD

**Issue by jiejiangd** [Engler93/Self-Supervised-Autogenous-Learning#1]
Would you like to ask how to generate.txt file

**Comment by Philipp-Engler** [comment]
You can generate them with the `create_grouping_v2(matrix, num_groups, group_similar, save_labels, model_name, size_margin=1.1, class_map=None)` function in either `helper.py` of the keras folder or `helper_pytorch.py` of the pytorch folder. 

- `matrix` is some confusion matrix of the classes as numpy array. It may be obtained evaluating any trained model. 
- `num_groups` is the number of groups that the classes should be split up to. 
- `group_similar` flag should be True, if you want often confused (~ similar) classes to end up in the same groups or False, if you want similar classes to end up in different groups. 
- `save_labels` needs to be True in order to save the resulting grouping as .txt file. 
- `model_name` could be any identifier and is put into the filename, but the code expects it to be `imagenet` or `cifar100`. 
- `size_margin` can be left at what it is to allow some slight variability in the group size or put to 1.0 in order to get even more balanced groups. 
- `class_map` is not really relevant. 

For the text file each line represents one class of the dataset and the value in that line represents the id of the group it belongs to. It also may be generated manually or using any other clustering algorithm as long as the file names match and `numpy.loadtxt` can load it correctly.
